### PR TITLE
Share arrays of added and removed addresses on ringChanged

### PR DESCRIPTION
**Pull request by kriskowal**
Over in Hyperbahn, I would like to incrementally update sorted arrays of affine relay addresses to reduce the cost of updating the service proxy in response to ring changes. To that end, I would like to thread the arrays of added and removed servers through the event.

I’ve added this feature and altered the existing test to verify the intended added and removed hosts, initializing the test to reflect reality since, but please check…it does not match my intuition.

r @uber/ringpop @jwolski

### CODE REVIEW

File: lib/on_membership_event.js [hunk 1]
```diff
@@ -62,7 +63,8 @@ function createSetHandler(ringpop) {
         // efficiency gains when only having to compute the ring
         // checksum once.
         if (serversToAdd.length > 0) {
-            ringpop.ring.addRemoveServers(serversToAdd);
+            ringpop.ring.addRemoveServers(serversToAdd, empty);
```
Comment (jcorbin) [inline lib/on_membership_event.js]: push down the var empty to here at least

**Comment by Raynos** [comment]
This seems reasonable to me.

File: test/unit/index-test.js [hunk 2]
```diff
@@ -394,18 +396,30 @@ test('emits ring changed event', function t(assert) {
...
+    }, {
+        adding: [],
+        removing: [node1Addr]
     });
 
     assertChanged(function assertIt() {
         ringpop.membership.makeAlive(node1Addr, magicIncNo);
+    }, {
+        adding: [],
+        removing: [node1Addr]
```
Comment (kriskowal) [inline test/unit/index-test.js]: This formulation makes the tests pass, but doesn’t match my intuition of what should happen in this case.

**Comment by kriskowal** [comment]
Fixed a bug and the test cases that pass make more sense now.

**Comment by jwolski** [comment]
@kriskowal I pushed some commits on top of yours to keep it up-to-date with current approaches. I introduced `RingChangedEvent`. Make sure you take note of that. Otherwise, let's wait for a passing test and then I'll merge it in if it looks good to you.

**Comment by kriskowal** [comment]
@jwolski This will get the job done. Thanks.

**Pull request merged**